### PR TITLE
Update codegen part 1

### PR DESCRIPTION
**Pull request by pawelprazak**
Taken from #258

### CODE REVIEW

File: Justfile [hunk 1]
```diff
@@ -220,7 +220,7 @@ package-language-plugins-all: build-language-plugin-bootstrap
 
 # Compiles codegen module
 compile-codegen:
-	scala-cli compile codegen
+	scala-cli --power compile codegen --suppress-experimental-feature-warning
```
Comment (prolativ) [inline Justfile:223]: `--power` shouldn't be needed here

File: Justfile [hunk 2]
```diff
@@ -231,17 +231,21 @@ get-schema schema-name schema-version:
...
 	pulumi package get-schema $schema_source > $schema_dir/schema.json
 
+# Publishes locally besom codegen
+publish-local-codegen:
+	scala-cli --power publish local codegen --project-version {{besom-version}} --suppress-experimental-feature-warning
+
 # Generate scala API code for the given provider, e.g. `just generate-provider-sdk kubernetes`
 generate-provider-sdk schema-name schema-version:
-	scala-cli run codegen -- {{schemas-output-dir}} {{codegen-output-dir}} {{schema-name}} {{schema-version}} {{besom-version}}
+	scala-cli --power run codegen --suppress-experimental-feature-warning -- {{schemas-output-dir}} {{codegen-output-dir}} {{schema-name}} {{schema-version}} {{besom-version}}
```
Comment (prolativ) [inline Justfile:240]: `--power` shouldn't be needed here

File: Justfile [hunk 3]
```diff
@@ -231,17 +231,21 @@ get-schema schema-name schema-version:
...
 
 # Compiles the previously generated scala API code for the given provider, e.g. `just compile-provider-sdk kubernetes`
 compile-provider-sdk schema-name:
-	scala-cli compile {{codegen-output-dir}}/{{schema-name}}
+	scala-cli --power compile {{codegen-output-dir}}/{{schema-name}} --suppress-experimental-feature-warning
```
Comment (prolativ) [inline Justfile:244]: `--power` shouldn't be needed here

File: codegen/project.scala [hunk 4]
```diff
@@ -1,4 +1,16 @@
-//> using scala "2.13.10"
-//> using lib "org.scalameta::scalameta:4.8.10"
-//> using lib "com.lihaoyi::upickle:3.1.3"
-//> using lib "com.lihaoyi::os-lib:0.9.1"
+//> using scala 2.13.12
+//> using options "-release:11", "-target:11"
```
Comment (prolativ) [inline codegen/project.scala:2]: `-target:11` should be redundant when `-release:11` is set

File: scripts/Schemas.scala [hunk 5]
```diff
@@ -43,8 +43,18 @@ private def fetchSchemas(cwd: os.Path): Unit =
   val targetPath = cwd / "integration-tests" / "resources" / "testdata"
   os.remove.all(targetPath)
 
+  // copy test schemas
   copySchemas(pulumiRepo / "pkg" / "codegen" / "testing" / "test" / "testdata", targetPath)
   copySchemas(pulumiJavaRepo / "pkg" / "codegen" / "testing" / "test" / "testdata", targetPath)
+
+  // fetch sample production schemas
+  List(
+    "kubernetes",
```
Comment (prolativ) [inline scripts/Schemas.scala:52]: There are already some schemas (including these providers) in pulumi repo testdata. Why don't we use these?

File: Justfile (hunk 1 above)
Comment (pawelprazak) [reply Justfile:223]: it will be in the next PR, when we publish the codegen module ;)

File: Justfile (hunk 2 above)
Comment (pawelprazak) [reply Justfile:240]: it will be soon, trust me ;)

File: Justfile (hunk 3 above)
Comment (pawelprazak) [reply Justfile:244]: it will be soon, trust me ;)

**Review by prolativ — APPROVED** [review]

**Pull request merged**